src/remotelog.lua

```diff
@@ -0,0 +1,221 @@
+        M.info("%sConnected to log receiver listening on %s:%d with log level %s. Time zone is UTC%s.", log_client_prefix, host, port,
+            get_level_name(M.level), os.date("%z"))
+    else
+        print(log_client_prefix .. "W-LOG-2: Unable to open socket connection to " .. host .. ":" .. port
+            .. " for sending log messages. Falling back to console logging with log level " .. get_level_name(M.level)
+            .. ". Timezone is UTC" .. os.date("%z") .. ". Caused by: " .. err)
+    end
+end
+
+---
+-- Close the connection to the remote log receiver.
+--
+function M.disconnect()
+    if(M.socket_client) then
+        M.socket_client:close()
+    end
+end
+
+---
+-- Set the log level.
+--
+-- @param level_name name of the log level, one of: FATAL, ERROR, WARN, INFO, CONFIG, DEBUG, TRACE
+--
+function M.set_level(level_name)
+    local level = levels[level_name]
+    if not level then
+        M.warn('W-LOG-1: Attempt to set illegal log level "%s". Pick one of: NONE, FATAL, ERROR, WARN, INFO, CONFIG,'
+        .. ' DEBUG, TRACE. Falling back to level INFO.', level_name)
+        M.level = levels.INFO
+    else
+        M.level = level
+    end
+end
+
+---
+-- Write to a socket, print or discard message.
+-- <p>
+-- If a socket connection is established, this method writes to that socket. Otherwise if the global print function
+-- exists (e.g. in a unit test) falls back to logging via <code>print()</code>.
+-- </p><p>
+-- Exasol removed print() in it's Lua implementation, so there is no fallback on a real Exasol instance. You either use
+-- remote logging or messages are discarded immediately.
+-- </p>
+--
+-- @param level log level
+--
+-- @param message log message; otherwise used as format string if any variadic parameters follow
+--
+-- @param ... parameters to be inserted into formatted message (optional)
+--
+local function write(level, message, ...)
+    if not (M.socket_client or print) then
+        return
+    else
+        local entry
+        local formatted_message = (select('#', ...) > 0) and string.format(message, ...) or message
+        if M.use_high_resolution_time then
+            local current_millis = string.format("%07.3f", (socket.gettime() - M.start_nanos) * 1000)
+            entry = {
+                os.date(M.timestamp_pattern),
+                " (", current_millis, "ms) [", level , "]",
+                string.rep(" ", 7 - string.len(level)), formatted_message
+            }
+        else
+            entry = {
+                os.date(M.timestamp_pattern),
+                " [", level , "]",
+                string.rep(" ", 7 - string.len(level)), formatted_message
+            }
+        end
+        if M.socket_client then
+            entry[#entry + 1] = "\n"
+            M.socket_client:send(table.concat(entry))
+        else
+            if print then
+                print(table.concat(entry))
+            end
+        end
+    end
+end
+
+function M.fatal(...)
```

Comment:
I guess these are public methods. So do they need doc?

Reply:
You are right. Added.